packages/core/src/browser/core-preferences.ts

```diff
@@ -72,6 +72,18 @@ export const corePreferenceSchema: PreferenceSchema = {
             A setting of 'compact' will move the menu into the sidebar.`,
             included: !isOSX
         },
+        'window.titleBarStyle': {
```

Comment:
In vscode, the preference also controls the display of context-menus, meaning we can have browser-like menus in electron instead of native. That's why when I first thought of the issue I was thinking we can control the binding we do to not bind native electron menus and instead use that of the browser:

https://github.com/eclipse-theia/theia/blob/068e78d34f88f274c9c7f9ddd64299fd7796589d/packages/core/package.json#L123-L124

With a bit more extra work, we can simply use the browser menus in electron (main menu, and context menus) by modifying the way we bind.

Reply:
Do we want to have browser-like menus in Electron always? I'm not quite sure how we would be able to accomplish that during the binding phase, as we need the preference service to identify which title bar style is selected.

Reply:
@msujew I believe dynamic binding might be possible, and is something we likely did in https://github.com/eclipse-theia/theia/pull/9186.

As for the behavior, I would expect that:
- `native` displays the native menus (main-menu and context-menus) similarly to vscode
- `custom` displays the browser-like menus (main-menu and context-menus) similarly to vscode

You're right that there might be issues especially on startup, but perhaps it's something to try.
cc @paul-marechal any thoughts?

Reply:
You reminded me that the context menus where not displayed in the browser styling if `custom` is used. I addressed that.

> I believe dynamic binding might be possible, and is something we likely did in #9186.

I only saw conditional binding based on `os` based constants. That works fairly well. However, during binding we have no access to preferences as they are 1. a service in themselves 2. we would have to wait for the async `ready` state.